nntrainer/layers/layer_context.cpp

```diff
@@ -274,9 +274,20 @@ const Tensor &RunLayerContext::getInput(unsigned int idx) const {
  * @return Tensor& Reference to the input grad tensor
  */
 Tensor &RunLayerContext::getInputGrad(unsigned int idx) {
-  if (!inputs[idx]->hasGradient())
-    throw std::invalid_argument(
-      "Requesting gradient for a non-trainable tensor.");
+  if (!inputs[idx]->hasGradient()) {
+    //
+    // TODO: This is temporal fixes. We need to find way to handle
+    //       the getInputGrad for multiple inputs ( input 1: Input Layer,
+    //       input 2: Normal Input ). In this situation, if we enable
+    //       inPlaceOptimization, it will fail.
+    //
+    // throw std::invalid_argument(
+    //   "Requesting gradient for a non-trainable tensor.");
+    //
+    Tensor var = inputs[idx]->getVariableRef();
+    Tensor t_grad = var;
+    inputs[idx]->initializeGradient(t_grad);
```

Comment:
This will make gradient share memory with the variable itself which can cause errors in certain cases.
I recommend to consider `return Tensor(inputs[idx]->getDim());`

Reply:
So, you mean we have to create tensor every time, then?

Reply:
> So, you mean we have to create tensor every time, then? 

yes, we would have to. The current implementation can create issues if the `input->getVariableRef()` is used later, as the gradient is sharing the memory and can overwrite the tensor data. 

However, as this is just a temporary fix, if its working, lets keep and add this as a note/comment here.

Reply:
fixed